android/src/com/mapswithme/maps/downloader/DownloaderAdapter.java

```diff
@@ -364,68 +311,98 @@ public void run()
 
     private void processLongClick()
     {
-      List<MenuItem> items = new ArrayList<>();
+      ArrayList<MenuBottomSheetItem> items = getMenuItems();
+      if (items.isEmpty())
```

Comment:
How is it possible? IMO, it is better to crash in "impossible" cases (so the core reason for a bug can be fixed earlier) than to silently return and ignore it.

Reply:
I'll throw an IllegalStateException instead of a silent return

Reply:
You can just don't check it, and there will be a null exception. YAGNI :)